site/app/views/NavigationView.php

```diff
@@ -242,21 +249,216 @@ public function showGradeables($sections_to_list) {
+                        $button_text = "OVERDUE SUBMISSION {$submission_status["SUBMITTED"]} {$submission_status["AUTOGRADE"]} {$display_date}";
+                    } //when there is no TA grade and due date passed
+                    else if($title_save == "GRADED" && $g_data->useTAGrading() && !$g_data->beenTAgraded()) { 
+                        $button_text = "TA GRADE NOT AVAILABLE {$submission_status["SUBMITTED"]} 
+                        	{$submission_status["AUTOGRADE"]} {$display_date}";
+                        $title_to_button_type_submission['GRADED'] = "btn-default";
+                    }
+                    else {
+                    	$button_text = "{$title_to_prefix[$title]} {$submission_status["SUBMITTED"]} {$submission_status["AUTOGRADE"]} {$display_date}";
+                    }
                     if ($g_data->hasConfig()) {
-                        $gradeable_open_range = <<<HTML
+                        //calculate the point percentage
+                    	if ($g_data->getTotalNonHiddenNonExtraCreditPoints() == 0) {
+                    		$points_percent = 0;
+                    	}
+                    	else {
+                    		$points_percent = $g_data->getGradedNonHiddenPoints() / $g_data->getTotalNonHiddenNonExtraCreditPoints();
+                    	}                    	
+						$points_percent = $points_percent * 100;
+						if ($points_percent > 100) { $points_percent = 100; }
+                        //prints out the button if $points_percent >= 50% or if there is no autograding points
+						if (($g_data->beenAutograded() && $g_data->getTotalNonHiddenNonExtraCreditPoints() != 0 && $g_data->getActiveVersion() >= 1
+							&& $title_save == "CLOSED" && $points_percent >= 50) || ($g_data->beenAutograded() && $g_data->getTotalNonHiddenNonExtraCreditPoints() == 0 && $g_data->getActiveVersion() >= 1)) {
+						$gradeable_open_range = <<<HTML
+                 <button class="btn btn-default" style="width:100%;" onclick="location.href='{$site_url}&component=student&gradeable_id={$gradeable}';">
+                     {$button_text}
+                 </button>
+HTML;
+						}
+						else { //prints out the button if $points_percent < 50%
+							$gradeable_open_range = <<<HTML
                  <button class="btn {$title_to_button_type_submission[$title]}" style="width:100%;" onclick="location.href='{$site_url}&component=student&gradeable_id={$gradeable}';">
                      {$button_text}
                  </button>
 HTML;
+						}
+                        
+						//If the button is autograded and has been submitted once, give a progress bar.
+						if ($g_data->beenAutograded() && $g_data->getTotalNonHiddenNonExtraCreditPoints() != 0 && $g_data->getActiveVersion() >= 1
+							&& ($title_save == "CLOSED" || $title_save == "OPEN"))
+						{							
+							//from https://css-tricks.com/css3-progress-bars/
+							if ($points_percent >= 50) { //if $points_percent >= 50%
```

Comment:
This comment (and the one on the else) don't really add any sort of clarification that isn't gleamed from just reading the if statement conditionals.